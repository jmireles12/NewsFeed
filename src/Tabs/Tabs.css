.Filter {
    display: flex;
    justify-content: center;
    padding-top: 10px;
}

.dropdown {
    align-items: center;
    width: 150px;
    border-radius: 5px;
    border-color: #004e98;
}

.edit__button {
    height: 25px;
    border-radius: 5px;
    border-width: thin;
}