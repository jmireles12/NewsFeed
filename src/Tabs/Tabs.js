import React, { Component } from 'react'
import { Link } from 'react-router-dom'
import ApiContext from '../ApiContext'
import 'react-tabs/style/react-tabs.css'
import './Tabs.css'

export default class Tabs extends Component {

    constructor(props) {
        super(props);
        this.state = {
            value: '1'
        }
    }

    static defaultProps = {
        match: {
            params: {}
        }
    }

    static contextType = ApiContext;


    handleChange = (e) => {
        let queryValue = '&q=' + e.target.value
        this.context.addQueryValue(queryValue)
        this.context.fetchData(queryValue)
        this.setState({ value: e.target.value})
    }


    render() {
        const { tabs=[] } = this.context
        return (
            <section className='Filter'>
                <select className='dropdown' value={this.state.value} name='filter-list' onChange={this.handleChange}>
                    <option value='1' disabled>...</option>
                    {tabs.map(tab =>
                        <option key={tab.id} value={tab.name}>
                            {tab.name}
                        </option>)}
                </select>
                <Link to='/add-tab'>
                    <button className='edit__button'>
                        Edit/Add
                    </button>    
                </Link>
            </section>
        )
    }
}
  
