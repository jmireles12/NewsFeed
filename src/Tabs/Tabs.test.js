import React from 'react'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import Tabs from './Tabs'

describe(`Tabs component`, () => {
    it('renders the complete Tabs', () => {
        const wrapper = shallow(<Tabs />)
        expect(toJson(wrapper)).toMatchSnapshot()
    })
})