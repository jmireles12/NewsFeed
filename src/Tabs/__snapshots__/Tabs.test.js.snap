// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tabs component renders the complete Tabs 1`] = `
<section
  className="Filter"
>
  <select
    className="dropdown"
    name="filter-list"
    onChange={[Function]}
    value="1"
  >
    <option
      disabled={true}
      value="1"
    >
      ...
    </option>
  </select>
  <Link
    to="/add-tab"
  >
    <button
      className="edit__button"
    >
      Edit
    </button>
  </Link>
</section>
`;
